Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h

```diff
@@ -0,0 +1,799 @@
+}
+
+void insertSparseMatrix(const SparseMat& mat, std::vector<SparseTriplet>& coefficients, size_t start_i = 0, size_t start_j = 0) {
+  for (int k = 0; k < mat.outerSize(); ++k)
+    for (SparseMat::InnerIterator it(mat, k); it; ++it)
+      coefficients.push_back(SparseTriplet(start_i + it.row(), start_j + it.col(), it.value()));
+}
+
+/*
+std::pair<Matrix, Vertex> point2point(Vertices X, Vertices Y) { // Transform step for rigid ICP
+  assert(X.rows() == Y.rows());
+
+  Eigen::Vector<ScalarType, 3> x_mean = X.colwise().mean();
+  X.rowwise() -= x_mean.transpose();
+
+  Eigen::Vector<ScalarType, 3> y_mean = Y.colwise().mean();
+  Y.rowwise() -= y_mean.transpose();
+
+  Matrix C = X.transpose() * Y;
+  Eigen::JacobiSVD<Matrix> svd(C, Eigen::ComputeFullU | Eigen::ComputeFullV);
+
+  Eigen::Vector<ScalarType, 3> sgnVec;
+  sgnVec << 1, 1, sign(svd.matrixU().determinant() * svd.matrixV().determinant());
+  Matrix R = svd.matrixV() * sgnVec.asDiagonal() * svd.matrixU().transpose();
+  Vertex t = y_mean - R * x_mean;
+
+  return std::make_pair(R, t);
+}*/
+
+template <typename T>
+Eigen::Matrix<T, 3, 3> rot(T a, T b, T c) {
+  T ca = cos(a), cb = cos(b), cc = cos(c);
+  T sa = sin(a), sb = sin(b), sc = sin(c);
+  Eigen::Matrix<T, 3, 3> R;
+  R << cb * cc, cc* sa* sb - ca * sc, ca* cc* sb + sa * sc,
+    cb* sc, ca* cc + sa * sb * sc, ca* sb* sc - cc * sa,
+    -sb, cb* sa, ca* cb;
+  return R;
+}
+
+} // namespace internal
+
+/*!
+* \ingroup PMP_registration_grp
+*
+* \brief calculates non-rigid transformation of a mesh onto a set of oriented points.
+*
+* A non-rigid ICP, iterative closest point, method based on
+* <A HREF="https://vgl.ict.usc.edu/Research/NonRigidRegistration/MODERN%20TECHNIQUES%20AND%20APPLICATIONS%20FOR%20REAL-TIME%20NON-RIGID%20REGISTRATION.pdf">a SIGGRAPH'16 Tutorial</A>.
+* The method uses a few correspondences between the source and the target for the rough alignment. The iterative closest point method
+* iteratively approaches the target by minimizing the distance between vertices of the source and points of the target.
+*
+* @tparam TriangleMesh a model of `MutableFaceGraph`.
+* @tparam PointRange a model of the `concept ForwardRange` whose value type is the point type.
+* @tparam VertexTranslationMap is a property map with `boost::graph_traits<TriangleMesh1>::%vertex_descriptor`
+*   as key type and a \cgal Kernel `Vector_3` as value type.
+* @tparam VertexRotationMap is a property map with `boost::graph_traits<TriangleMesh1>::%vertex_descriptor`
+*   as key type and a \cgal Kernel `Aff_transformation_3` as value type.
+* @tparam NamedParameters1 a sequence of \ref bgl_namedparameters "Named Parameters"
+* @tparam NamedParameters2 a sequence of \ref bgl_namedparameters "Named Parameters"
+*
+* @param source the triangle mesh that should be mapped onto target.
+* @param target the target triangle mesh.
+* @param vtm a writeable vertex property map of source to store the translation vector of the registration.
```

Comment:
```suggestion
* @param vtm a writeable vertex property map of  `source` to store the translation vector of the registration.
```
similar for other usages of parmeters